### PR TITLE
Add ios support, Add audio support

### PR DESCRIPTION
**Pull request by czoido**

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -16,19 +25,30 @@ add_executable(mygame main.cpp
...
 target_include_directories(mygame PRIVATE include)
 
 # FIXME: sdl_image conan package still not adding properties
-target_link_libraries(mygame CLI11::CLI11 fmt::fmt box2d::box2d SDL2::SDL2main sdl_image::sdl_image tinkerforge::bindings)
+target_link_libraries(mygame fmt::fmt box2d::box2d SDL2::SDL2main sdl_image::sdl_image SDL2_mixer::SDL2_mixer tinkerforge::bindings)
+
+if(APPLE)
+        target_link_libraries(mygame -framework CoreServices -framework ImageIO)
+else()
+        target_link_libraries(mygame CLI11::CLI11)
```
Comment (danimtb) [inline src/CMakeLists.txt:41]: isn't this target needed in MacOS as well?

**Review by danimtb — APPROVED** [review]

File: .github/workflows/create_tag_release.yml [hunk 2]
```diff
@@ -33,7 +33,8 @@ jobs:
       library_name: ${{ github.event.repository.name }}
       library_version: ${{ needs.cmakelists-version.outputs.LIBRARY_VERSION }}
       github_sha: master
-      profile: release
+      profile_host: release
+      profile_build: release
```
Comment (jgsogo) [inline .github/workflows/create_tag_release.yml:37]: No need to change anything here. (Even if there were that parameter,) there is no need to force the `profile:build`, it is provided by the worker so it matches its own architecture.

```suggestion
      profile: release
```

File: .conan/profiles/ios_simulator [hunk 3]
```diff
@@ -0,0 +1,6 @@
+include(default)
+[settings]
+os=iOS
+os.sdk=iphonesimulator
+os.version=9.0
+arch=x86_64
```
Comment (jgsogo) [inline .conan/profiles/ios_simulator:6]: This is equal to the existing `ios` profile, just rename it

File: conanfile.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        self.requires("box2d/2.4.1")
+        self.requires("fmt/8.1.1")
+        self.requires("tinkerforge-bindings/2.1.32")
+
+        # Testing
+        self.requires("catch2/2.13.7")
+
+        # Overrides
+        self.requires("zlib/1.2.12", override=True)
+        self.requires("libxml2/2.9.13", override=True)
```
Comment (jgsogo) [inline conanfile.py:28]: What are the conflicts here? Can you open the required PRs to `conan-center-index`? Thanks!

File: src/main.cpp [hunk 5]
```diff
@@ -1,35 +1,39 @@
 #include "Game.hpp"
+#include "Platform.hpp"
 
+#ifdef IS_DESKTOP
 #include <CLI/CLI.hpp>
-#include "fmt/core.h"
+#endif
```
Comment (jgsogo) [inline src/main.cpp:6]: In the `conanfile.py` this CLI11 is added if `!ios`. I'm suggesting there to use something aligned with this logic (see comment above)

File: conanfile.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "CMakeToolchain", "CMakeDeps"
+
+    def configure(self):
+        if self.settings.os != "Linux":
+            self.options["sdl_mixer"].nativemidi = False
+
+    def requirements(self):
+        if self.settings.os != "iOS":
+            self.requires("cli11/2.1.2")
```
Comment (jgsogo) [inline conanfile.py:15]:
```suggestion
    @property
    def _is_desktop(self):
        return self.settings.os not in ["iOS", ]
 
    def requirements(self):
        if self._is_desktop:
            self.requires("cli11/2.1.2")
```

File: conanfile.py (hunk 4 above)
Comment (czoido) [reply conanfile.py:28]: No idea, this comes from your changes: https://github.com/froglets/flappy-frog/blame/905683f3b6e8855f9a834dc1ac7ee0b09a88fbc3/conanfile.txt#L12
I just ported that to `.py`.

Comment (jgsogo) [reply conanfile.py:32]: Here it goes: https://github.com/froglets/flappy-frog/pull/17

**Comment by jgsogo** [comment]
...and more conflicts

**Pull request closed without merging**